workflow/Snakefile

```diff
@@ -27,27 +27,23 @@ VERSION = ".".join(FULL_VERSION.split(".")[:2])
 if config.get("full-version", False):
     VERSION = FULL_VERSION
 
-
 # thread options
 MAX_THREADS = config.get("max_threads", 4)
 SORT_THREADS = config.get("sort_threads", 8)
 
+# sample, haplotype, and chromosome wildcard building
+DEFAULT_ENV = config.get("env", "../envs/env.yaml")
+MANIFEST = get_manifest()
```

Comment:
Add here a 
```
CHROM_DF=read_bam_headers_for_chroms()
```
This will be the only time we read all the bam headers and store, sample, chrom, and length. We can also apply the chromosome filtering here in the step. (e.g. removing chr_unk)

All other functions will the just pull from CHROM_DF